### PR TITLE
Updates docs for the igxAvatarComponent API

### PR DESCRIPTION
**Pull request by 3phase**
### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [x] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [ ] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/avatar/avatar.component.ts [hunk 1]
```diff
@@ -164,8 +182,9 @@ export class IgxAvatarComponent implements OnInit, AfterViewInit {
     public roundShape = false;
 
     /**
-     * Sets the avatar's `initials`/`icon` color.
+     * Sets the color of the avatar's initials' icon.
```
Comment (damyanpetev) [inline projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: initials and icon are 2 separate types of avatar

File: projects/igniteui-angular/src/lib/avatar/avatar.component.ts [hunk 2]
```diff
@@ -82,8 +94,9 @@ export class IgxAvatarComponent implements OnInit, AfterViewInit {
     protected iconTemplate: TemplateRef<any>;
 
     /**
-     * Returns the `aria-label` of the avatar.
+     * Returns the aria-label attribute of the avatar.
```
[the remark this answers is not in the document]
Comment (3phase) [reply projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: Not really. There was too many of the code blocks and I was following the pattern of the changes, applied beyond. I'd probably encircle all the attributes with `s again though.

**Comment by 3phase** [comment]
> @3phase Also (new addition to the wiki) - can you drop all `@memberof IgxAvatarComponent` from the property descriptions.

Done, thanks

File: projects/igniteui-angular/src/lib/avatar/avatar.component.ts [hunk 3]
```diff
@@ -227,6 +213,7 @@ export class IgxAvatarComponent implements OnInit, AfterViewInit {
     @Input()
     public src: string;
 
+    private _size: string | Size = 'small';
```
Comment (rkaraivanov) [inline projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: Maybe we can collapse the union just to `Size` type

Comment (3phase) [reply projects/igniteui-angular/src/lib/avatar/avatar.component.ts]: It's done

**Comment by damyanpetev** [comment]
@3phase @rkaraivanov Restored the removed size test asset as it does test a valid code branch. Plus, there's no type safety for `@Input`.

Also added a friendly name to the `src` attribute per @SAndreeva's suggestion. Should we also add a friendly name for the component itself?

PS: [Wiki section on meta tags ](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines#metadata-tags )has been expanded

**Comment by damyanpetev** [comment]
@rkaraivanov Slight issue with the simplification of the enum `@Input`-s. In TypeScript, unlike numeric values 🤷‍♂, string enum values are intentionally opaque (see https://github.com/microsoft/TypeScript/issues/17690#issuecomment-321319291), so this could break existing code that assigned strings to those props.

Also forgot there is also type check option ([strictTemplates](https://angular.io/guide/angular-compiler-options#stricttemplates)), and since assigning an actual enum in templates is rather cumbersome, the common way of using the value (` size="medium" `) will also fail type checks. Related: https://github.com/angular/angular/issues/35363

---
_We might have a misalignment of what we're trying to get out of enums and what they're intended to do. The point is to always refer to the public API (the enum members) so we avoid repeating strings (that are supposed to be free to change), yet we are very much stuck referring to the actual values for templates - kind of like the cake and eating it deal._

**Comment by ChronosSF** [comment]
@damyanpetev , @rkaraivanov , could you please reach an agreement as to what to do with this change?

**Comment by damyanpetev** [comment]
@ChronosSF I'd air on the side of caution and not break the current type.
Will continue to monitor the feature suggestion in Angular, though I doubt there's much that can be done with enums that are not meant to be interchangeable with string values. Might be better served with a string union type, but we'll need to look into that and definitely not the job of this PR :)

**Comment by 3phase** [comment]
@damyanpetev @ChronosSF @rkaraivanov the type of the avatar size is returned to its initial state now. Are there any other things to do?

**Pull request merged**